PowershellTools.Common/ServiceManagement/IntelliSenseContract/IPowershellIntelliSenseService.cs

```diff
@@ -5,11 +5,11 @@ namespace PowerShellTools.Common.ServiceManagement.IntelliSenseContract
     /// <summary>
     /// Powershell service.
     /// </summary>
-    [ServiceContract]
+    [ServiceContract(CallbackContract = typeof(IIntelliSenseServiceCallback))]
     public interface IPowershellIntelliSenseService
     {
         [OperationContract]
-        CompletionResultList GetCompletionResults(string scriptUpToCaret, int carePosition);
+        void RequestCompletionResults(string scriptUpToCaret, int carePosition, string triggerTimeStamp);

```

Comment:
DateTime's not serializable?

If not: Any reason we don't send it over the wire as ticks to avoid the conversion back/forth from string?